From 79e70b40bb861889c337f15850ebc78b43545089 Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Mon, 19 Feb 2018 19:37:08 +0700
Subject: [PATCH] Problem: sit init does not respect `-r/--repository`

It will try to initialize a `.sit` repository

Solution: make it respect supplied repository path
---
 sit/src/main.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sit/src/main.rs b/sit/src/main.rs
index b0c8990..05c6d2d 100644
--- a/sit/src/main.rs
+++ b/sit/src/main.rs
@@ -281,7 +281,7 @@ fn main_with_result() -> i32 {
 
     if let Some(init_matches) = matches.subcommand_matches("init") {
         let dot_sit_str = matches.value_of("repository").unwrap_or(dot_sit.to_str().unwrap());
-        match sit_core::Repository::new(&dot_sit) {
+        match sit_core::Repository::new(&dot_sit_str) {
             Ok(repo) => {
                 if !init_matches.is_present("dont-populate") {
                     repo.populate_default_files().expect("can't populate default files");
-- 
2.15.0

